src/qinfer/geometry.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+##
+# geometry.py: Computational geometry utilities.
+##
+# © 2016 Chris Ferrie (csferrie@gmail.com) and
+#        Christopher E. Granade (cgranade@gmail.com)
+#     
+# This file is a part of the Qinfer project.
+# Licensed under the AGPL version 3.
+##
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+##
+
+## FEATURES ###################################################################
+
+from __future__ import absolute_import
+from __future__ import division
+
+## EXPORTS ###################################################################
+
+__all__ = [
+    'convex_distance',
+    'approximate_convex_hull'
+]
+
+## IMPORTS ###################################################################
+
+import numpy as np
+from scipy.spatial.distance import cdist
+
+from qinfer.utils import requires_optional_module
+
+try:
+    import cvxopt
+    import cvxopt.solvers
+except:
```

Comment:
Probably, though I don't want any error in importing CVXOPT to affect functions which don't require it. Perhaps silently ignoring ``ImportError`` and raising a warning on other exceptions would be good?